File: cobalt/frontend/src/types/schemas/games.ts
/*
 * Copyright (C) 2026 Artorias
 * Author: Artorias
 * Repository: https://github.com/artorias-developer/cobalt
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */

import type { SortDirection } from "@/types"

/**
 * Represents a paginated games list request.
 */
export interface GamesPageRequest {
  page?: number
  search?: string
  sort_field?: "id" | "name" | "loader_id" | "created_at" | "updated_at"
  sort_direction?: SortDirection
  limit?: number
}

File: cobalt/backend/domain/enums/games.py
#  Copyright (C) 2026 Artorias
#  Author: Artorias
#  Repository: https://github.com/artorias-developer/cobalt
#  SPDX-License-Identifier: AGPL-3.0-or-later

from enum import StrEnum


class GameSortFieldEnum(StrEnum):
    ID = "id"
    NAME = "name"
    LOADER_ID = "loader_id"
    CREATED_AT = "created_at"
    UPDATED_AT = "updated_at"